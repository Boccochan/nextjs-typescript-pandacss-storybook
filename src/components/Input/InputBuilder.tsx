import type { ComponentProps } from "react";
import React from "react";

import { InputBase } from ".";
import type { InputBuilderVariants } from "./InputBuilder.styles";
import { styles } from "./InputBuilder.styles";

type BasicProps = ComponentProps<typeof InputBase>;

type InputBuilderProps = Omit<BasicProps, "size"> & {
  /**
   * The size of the component.
   */
  size?: InputBuilderVariants["size"];

  /**
   * The width of input. The width of the Input component can be fixed.
   */
  width?: InputBuilderVariants["width"];
};

export const InputBuilder = React.forwardRef<HTMLDivElement, InputBuilderProps>(
  ({ size, width, ...rest }, ref) => {
    return (
      <div ref={ref} className={styles.inputBuilder({ size, width })}>
        <InputBase {...rest} size={size} />
      </div>
    );
  },
);

InputBuilder.displayName = "InputBuilder";
