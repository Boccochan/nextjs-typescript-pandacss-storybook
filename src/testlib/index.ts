export * from "./render";
export * from "./translator";
