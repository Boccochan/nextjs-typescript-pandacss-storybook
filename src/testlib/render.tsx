import { render as rtlRender } from "@testing-library/react";
import { NextIntlClientProvider } from "next-intl";
import React from "react";

import messageEn from "../../messages/en.json";
import messageJa from "../../messages/ja.json";

type RenderTuple = Parameters<typeof rtlRender>;

const messages = {
  en: messageEn,
  ja: messageJa,
};

export const render = (
  ui: RenderTuple[0],
  options?: RenderTuple[1],
  local?: "en" | "ja",
) =>
  rtlRender(ui, {
    wrapper: ({ children }) => (
      <NextIntlClientProvider
        locale={local ?? "en"}
        messages={local ? messages[local] : messageEn}
      >
        {children}
      </NextIntlClientProvider>
    ),
    ...options,
  });
