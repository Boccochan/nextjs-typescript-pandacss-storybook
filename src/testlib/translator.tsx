import { createTranslator } from "next-intl";

import { locales } from "../i18n";

export const getTranslator = async (locale: string) => {
  if (!locales.includes(locale)) throw Error("Unkown locale");
  const messages = (await import(`../../messages/${locale}.json`)).default;

  return createTranslator({ locale, messages });
};
