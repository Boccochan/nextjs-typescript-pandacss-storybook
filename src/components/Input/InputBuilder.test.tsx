import React from "react";

import { render } from "@/testlib";

import { InputBuilder } from "./InputBuilder";

describe("Snapshot InputBuilder", () => {
  it("With default", async () => {
    const inputBuilder = render(<InputBuilder />);

    expect(inputBuilder).toMatchSnapshot();
  });
});
