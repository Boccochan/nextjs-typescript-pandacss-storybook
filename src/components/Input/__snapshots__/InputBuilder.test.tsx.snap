// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot InputBuilder With default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border_1px_solid_{colors.body.border} bg_body.bg text_body.text d_flex justify_center items_center focusWithin:ring_2px_solid_black focusWithin:osDark:ring_2px_solid_white rounded_xs min-w_14rem h_2.4rem w_14rem"
      >
        <div
          class="h_100% px_2 d_flex justify_center items_center text_inputIcon fs_2xl"
        />
        <input
          class="border_none bg_body.bg text_body.text ring_none disabled:opacity_0.6 disabled:bg_body.readonly readOnly:pointer-events_none readOnly:bg_body.readonly d_block w_100% fs_md"
        />
        <div
          class="h_100% px_2 d_flex justify_center items_center text_inputIcon fs_2xl"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border_1px_solid_{colors.body.border} bg_body.bg text_body.text d_flex justify_center items_center focusWithin:ring_2px_solid_black focusWithin:osDark:ring_2px_solid_white rounded_xs min-w_14rem h_2.4rem w_14rem"
    >
      <div
        class="h_100% px_2 d_flex justify_center items_center text_inputIcon fs_2xl"
      />
      <input
        class="border_none bg_body.bg text_body.text ring_none disabled:opacity_0.6 disabled:bg_body.readonly readOnly:pointer-events_none readOnly:bg_body.readonly d_block w_100% fs_md"
      />
      <div
        class="h_100% px_2 d_flex justify_center items_center text_inputIcon fs_2xl"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
